cmd/migration-tool/main.go

```diff
@@ -0,0 +1,524 @@
+		klog.V(4).Infof("sourceImageStream: %s", sourceImageStream)
+
+		parts := strings.Split(sourceImageStream, "/")
+		if len(parts) != 2 {
+			klog.Warningf("Unable to determine the namespace/name for %s: %s", tag.Name, sourceImageStream)
+			skipped = append(skipped, tag.Name)
+			continue
+		}
+		sourceImageStreamNamespace := parts[0]
+		sourceImageStreamName := parts[1]
+
+		if c.Namespace != sourceImageStreamNamespace {
+			klog.Errorf("Validation error: inconsistent namespaces: %s != %s", c.Namespace, sourceImageStreamNamespace)
+			problems = append(problems, tag.Name)
+			continue
+		}
+
+		status, err := getAnnotation(tag, releasecontroller.ReleaseAnnotationVerify, true)
+		if err != nil {
+			klog.Warningf("Unable to determine verification status annotation: %v", err)
+			skipped = append(skipped, tag.Name)
+			continue
+		}
+
+		verifyStatus := make(releasecontroller.VerificationStatusMap)
+		if err = json.Unmarshal([]byte(status), &verifyStatus); err != nil {
+			klog.Warningf("Unable to unmarshal verification status for %s: %v", tag.Name, err)
+			skipped = append(skipped, tag.Name)
+			continue
+		}
+
+		message, err := getAnnotation(tag, releasecontroller.ReleaseAnnotationMessage, false)
+		if err != nil {
+			klog.Warningf("Unable to determine message annotation: %v", err)
+		}
+		klog.V(4).Infof("          message: %s", message)
+
+		reason, err := getAnnotation(tag, releasecontroller.ReleaseAnnotationReason, false)
+		if err != nil {
+			klog.Warningf("Unable to determine reason annotation: %v", err)
+		}
+		klog.V(4).Infof("           reason: %s", reason)
+
+		releases = append(releases, MigrationData{
+			Name:                       tag.Name,
+			ReleaseStream:              releaseStream,
+			Phase:                      phase,
+			SourceImageStreamName:      sourceImageStreamName,
+			SourceImageStreamNamespace: sourceImageStreamNamespace,
+			Status:                     verifyStatus,
+			Message:                    message,
+			Reason:                     reason,
+		})
+	}
+
+	return releases, skipped, problems
+}
+
+func getAnnotation(tag v1.TagReference, annotation string, required bool) (string, error) {
+	var ok bool
+	var content string
+	if content, ok = tag.Annotations[annotation]; !ok {
+		if required {
+			return "", fmt.Errorf("unable to locate '%s' annotation in tag: %s", annotation, tag.Name)
+		}
+		content = ""
+	}
+	return content, nil
+}
+
+func (c Controller) generatePayloads(releases []MigrationData, configs map[string]*releasecontroller.ReleaseConfig) []*v1alpha1.ReleasePayload {
+	var payloads []*v1alpha1.ReleasePayload
+	for _, release := range releases {
+		config := configs[fmt.Sprintf("%s/%s", release.SourceImageStreamNamespace, release.SourceImageStreamName)]
+		payloads = append(payloads, newReleasePayload(release, config, c.ProwNamespace))
+	}
+	return payloads
+}
+
+func newReleasePayload(release MigrationData, config *releasecontroller.ReleaseConfig, prowNamespace string) *v1alpha1.ReleasePayload {
```

Comment:
you need a canonicalize method that takes your ReleasePayload struct to have a consistent ordering of the various jobs for comparison purposes.